dbt/adapters/athena/python_submissions.py

```diff
@@ -162,18 +144,48 @@ def submit(self, compiled_code: str) -> Any:
                     "Result"
                 ]
             except Exception as e:
-                LOGGER.error(f"Unable to poll execution status: Got: {e}")
+                LOGGER.error(f"Unable to retrieve results: Got: {e}")
                 result = {}
-        self.spark_connection.release_session_lock(self.session_id)
         return result
 
+    def poll_until_session_idle(self) -> None:
+        """
+        Polls the session status until it becomes idle or exceeds the timeout.
+
+        Raises:
+            DbtRuntimeError: If the session chosen is not available or if it does not become idle within the timeout.
+        """
+        polling_interval = self.polling_interval
+        while True:
+            timer = 0
+            session_status = self.get_current_session_status()["State"]
+            if session_status in ["TERMINATING", "TERMINATED", "DEGRADED", "FAILED"]:
+                LOGGER.debug(
+                    f"The session: {self.session_id} was not available. Got status: {session_status}. Will try with a different session."
+                )
+                self.spark_connection.remove_terminated_session(self.session_id)
+                if "session_id" in self.__dict__:
+                    del self.__dict__["session_id"]
```

Comment:
concern: I think once this property is deleted I think we might have to raise an error and fail the model. Else we somehow have to assign a new session and retry which I think might be complicated. Because otherwise I'm not sure how this model will succeed

Reply:
looks like this is a while true loop and it keeps checking for the session id
if session id doesnt exist it is attempting to get a new session id 
so deleting means to grab a new session and cache that
i actually made a couple big changes with regards to session creation and threads
let me add a comment with the detailed explaination